### PR TITLE
Add Data Attributes for Automatic Newsletter Template Generation

### PR DESCRIPTION
**Pull request by jayjayjpg**
## What it does

Part of https://github.com/ember-learn/ember-times-tools/pull/28

This adds a couple of unique data-* attributes to parts of the Ember Times posts to facilitate the automated newsletter template generation process.

### CODE REVIEW

File: source/javascripts/embertimes-test-ids.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+   function applyEmberTimesTestIds() {
+     let sectionTitles = $('[data-embertimes-post-body] h2');
+     sectionTitles.attr('data-embertimes-section-title', true);
+
+     sectionTitles.each((i, title) => {
+       $(title).attr('data-embertimes-section-title', i);
+       let sectionFragments = $(title).nextUntil('h2').not('hr');
+       sectionFragments.each((index, element) => {
+         $(element).attr('data-embertimes-section-fragment', i);
```
Comment (ijlee2) [inline source/javascripts/embertimes-test-ids.js:20]: Should this be:
```suggestion
         $(element).attr('data-embertimes-section-fragment', index);
```
If you need to distinguish all section fragments, I might combine IDs `${i}_${index}` (can't get backtics to show up, to indicate string interpolation).

**Comment by ijlee2** [comment]
@jessica-jordan I think I got the gist of what you wanted to do, and that looks ok. For maintainability, I think adding comments to describe what you thought was `title`, what was an `intro fragment`, and what was a `section fragment` could help. Because you defined the fragments using `nextUntil` and
`not`, I wasn't quite sure which text in an Ember Times issue you wanted to grab.

There's one possible bug. Could you take a look?

File: source/javascripts/embertimes-test-ids.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+$(function(){
+  if ($('[data-embertimes-page]').length > 0) {
+    applyEmberTimesTestIds();
+  }
+
+
+   function applyEmberTimesTestIds() {
+     let sectionTitles = $('[data-embertimes-post-body] h2');
+     sectionTitles.attr('data-embertimes-section-title', true);
```
Comment (ijlee2) [inline source/javascripts/embertimes-test-ids.js]: Do we need this line? It seems, in line 12, you are setting a unique ID to section title.

Comment (jayjayjpg) [reply source/javascripts/embertimes-test-ids.js]: You're right, this one can be removed - thank you for the catch :+1:

Comment (jayjayjpg) [reply source/javascripts/embertimes-test-ids.js]:
```suggestion
```

File: source/javascripts/embertimes-test-ids.js (hunk 1 above)
Comment (jayjayjpg) [reply source/javascripts/embertimes-test-ids.js:20]: I think for the collection of the sections with the script it's mostly important that all the parts that a section is composed of can be identified as belonging together to that one section. E.g. each section may have several `p` or `ul` elements that we want to group together and copy-paste into one single textfield in the goodbits template interface in the end

I was thinking to just use the id that identifies a section to make sure we can easily [fetch all the sub parts of the section in one go in the automated template generation script](https://github.com/ember-learn/ember-times-tools/pull/28/files#r293800115) and I thought being able to distinguish the separate sub paragraphs wouldn't be relevant for the current template creation workflow for now

Does this make sense? Feel free to let me know if I missed something here

**Comment by jayjayjpg** [comment]
> For maintainability, I think adding comments to describe what you thought was title, what was an intro fragment, and what was a section fragment could help. Because you defined the fragments using nextUntil and
not, I wasn't quite sure which text in an Ember Times issue you wanted to grab.

@ijlee2 That's a great point, I'll add some commentary in a bit!

File: source/javascripts/embertimes-test-ids.js (hunk 1 above)
Comment (ijlee2) [reply source/javascripts/embertimes-test-ids.js:20]: Yep, I see what you wanted to do now. Thanks!

**Review by ijlee2 — APPROVED** [review]
I'll go ahead and approve this. Thanks for your work!

**Comment by amyrlam** [comment]
Nice, this is clever to add on this end... pumped to try it out. Thanks for making this change! 💯

**Pull request merged**